sematic/ui/src/types/Types.tsx

```diff
@@ -817,11 +861,25 @@ const TypeComponents: Map<string, ComponentPair> = new Map([
       value: DataFrameValueView,
     },
   ],
+  [
+    "sematic.types.types.aws.s3.S3Location",
+    {
+      type: TypeView,
+      value: S3LocationValueView,
+    }
+  ],
+  [
+    "sematic.types.types.aws.s3.S3Bucket",
+    {
+      type: TypeView,
+      value: S3BucketValueView,
+    }
+  ]
 ]);
 
 function getComponentPair(typeRepr: TypeRepr) {
   let typeKey = typeRepr[1];
-  if (typeRepr[0] === "class") {
+  if ("import_path" in typeRepr[2]) {
```

Comment:
This makes it so we don't have to create a shadow dataclass to have a custom viz for a particular dataclass. We can just register the React component with the full import path of the dataclass.

Reply:
Please add this as a code comment.

Reply:
Done.